### ISSUE TITLE
Unified search for file names and content

### ISSUE THREAD

**Issue by madduck** [eikek/docspell#2324]
On the Items search page, the search field defaults to searching in document names:

![image](https://github.com/eikek/docspell/assets/195073/06b551c9-7862-457f-b3a5-758f02ec71a0)

Everyone here assumed that the button next to the search field with the two arrows is the "Go" button, and we were surprised to find it switching the search mode to content:

![image](https://github.com/eikek/docspell/assets/195073/8040b117-7d18-474d-86b9-c9997becd965)

Is there a reason to differentiate? Why shouldn't document names *and* full-text contents be included in search results? After all, I don't do a web search for titles and content separately either.

If the two need to stay separate, can we maybe get two input fields, to make it clearer?

**Comment by eikek** [comment]
Yes this is indeed very confusing. The reason is that fulltext search is optional. You could run docspell without and the ui must somehow address this. The "names search" uses SQL only, where "content search" uses whatever fulltext engine is configured. The combination is the "advanced" search, which is behind a toggle in the user settings. This is only an explanation for how it works currently - not an excuse for it 😄 Should definitely be improved! 

We could have a single search that combines fulltext and SQL search. But the downside is that when fulltext search is used, the order is not by date but by relevance. So a user might be surprised by that. When the advanced search option is on, there is only one input field. I'm open for suggestions on how to make this better! I think in the past there was once a dropdown to select the different search modes… not sure perhaps that was better. But it is still confusing. 🤷🏼

**Comment by madduck** [comment]
@eikek maybe the easiest could be to forget about SQL-only title search, but let the full-text search engine index the titles as well?

**Comment by eikek** [comment]
The fulltext engine does index document and item names. But docspell can nicely run without any fulltext engine at all. The sql search is not as powerful, but much simpler. The biggest difference is that with sql search, you get a order by date and with fulltext search the order is by relevance.

**Comment by madduck** [comment]
So why not just show only the FTS search if FTS is configured, and only the SQL title search otherwise?

**Comment by eikek** [comment]
Sure, that is fine to me. Only the ordering implication exists then.

**Comment by madduck** [comment]
I would argue that this is an issue that could be resolved client-side with UI sort controls: return sorted by relevance, but let the user quickly resort with the column headers if desired.

**Comment by eikek** [comment]
> return sorted by relevance, but let the user quickly resort with the column headers if desired.

Yes, this must be first supported by the backend 😏 I think there is an issue about it already

**Comment by madduck** [comment]
I'll search a bit, but how is this backend relevant? i think it is just client side..

**Comment by eikek** [comment]
The backend currently cannot sort differently. There is no way to define a sort property - you always get back the same order currently.

**Comment by madduck** [comment]
So let's sort in the frontend… but I understand that if there are tons of search results and you only want to render them in batches, then the backend needs to do the sorting…

**Comment by eikek** [comment]
The issue I remembered is #1453 . Yes, sorting on the client doesn't work. It will mess up with with the "load more" button. It must be done at the server (and handed down to the db).
